optimade/server/routers/references.py

```diff
@@ -0,0 +1,57 @@
+from typing import Union
+
+from fastapi import APIRouter, Depends
+from starlette.requests import Request
+
+from optimade.models import (
+    ErrorResponse,
+    ReferenceResource,
+    ReferenceResponseMany,
+    ReferenceResponseOne,
+)
+from optimade.server.config import CONFIG
+from optimade.server.deps import EntryListingQueryParams, SingleEntryQueryParams
+from optimade.server.entry_collections import MongoCollection, client
+from optimade.server.mappers import ReferenceMapper
+
+from .utils import get_entries, get_single_entry
+
+
+router = APIRouter()
+
+references_coll = MongoCollection(
+    client[CONFIG.mongo_database]["references"], ReferenceResource, ReferenceMapper
+)
+
+
+@router.get(
+    "/references",
+    response_model=Union[ReferenceResponseMany, ErrorResponse],
+    response_model_skip_defaults=True,
```

Comment:
Getting lots of warnings about `resposne_model_skip_defaults` from the new FastAPI, could be worth updating it here to expedite the upgrade to pydantic>1.

```response_model_skip_defaults has been deprecated in favor of response_model_exclude_unset to keep in line with Pydantic v1, support for it will be removed soon.```

Reply:
Note: This keyword _only_ works for `fastapi>=0.44.0`. So we would need to pin this version as a minimum version if we use this keyword.

Reply:
That's fair, happy to leave it for big deps update PR then